app/scripts/metamask-controller.js

```diff
@@ -1394,6 +1395,27 @@ export default class MetamaskController extends EventEmitter {
 
     ///: END:ONLY_INCLUDE_IF
 
+    // Notification Controllers
+    // This controller relies on work from another PR.
+    // https://github.com/MetaMask/metamask-extension/pull/23286
+    // For now the methods have been stubbed out.
+    // NOTE - this controller is not connected to anything live yet & will only be functioning once the full feature is merged.
+    const stubbedAuthMethod = () => {
+      throw new Error('Unimplemented Method');
+    };
+    this.userStorageController = new UserStorageController({
+      messenger: this.controllerMessenger.getRestricted({
+        name: 'UserStorageController',
+        allowedActions: [`${this.snapController.name}:handleRequest`],
+      }),
+      auth: {
+        getBearerToken: stubbedAuthMethod,
+        getSessionIdentifier: stubbedAuthMethod,
+        isAuthEnabled: stubbedAuthMethod,
```

Comment:
Which function in the auth controller will be passed to `isAuthEnabled`?

Reply:
Yep! The auth controller has this property in its state: `isSignedIn: boolean`.
So once it is added we can provide it like:
```ts
// pseudo-code, might need async/await or handle promises, but will be similar to this.
auth: {
  getBearerToken: () => authController.getBearerToken(),
  getSessionIdentifier: () => authController.getSessionProfile()?.profileId,
  isAuthEnabled: () => authController.state.isSignedIn,
  signIn: () => authController.performSignIn()
}
```